server.go

```diff
@@ -282,12 +282,15 @@ func (s *Server) sendResponse(t transport.ServerTransport, stream *transport.Str
 }
 
 func (s *Server) processUnaryRPC(t transport.ServerTransport, stream *transport.Stream, srv *service, md *MethodDesc) (err error) {
+	ctx, cancel := context.WithCancel(stream.Context())
```

Comment:
This seems an odd place to introduce this, no? Shouldn't whoever is creating that context be canceling it? Anything that gets to the `transport.Stream` will be getting a context differing from this one.